### PR TITLE
[pickers] Support invalid dates from the field

### LINKED ISSUE #8296: [pickers] Unfocused field appears blank when partial date input is provided
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:
This can be reliably reproduced with any of the desktop variant examples: https://mui.com/x/react-date-pickers/date-picker/

Steps:
1. Populate the desktop date picker field with a valid date.
2. Delete any part of the date to make the input incomplete. Eg: 04 / DD / 2020
3. Unfocus the field - Notice the field goes blank, even though the date is partially provided.

### Current behavior 😯

When unfocused, the input field hides the partially populated date.
The issue only seems to happen after the field has first been filled with a full and valid date. It works as expected if the field hasn't previously been populated with a valid date.
Focusing the field again causes the partially filled date to re-appear.
I tested the DateField (https://mui.com/x/react-date-pickers/date-field/) following the same steps to reproduce and it doesn't have the issue.

### Expected behavior 🤔

Partially filled dates like "04 / DD / 2020" should always be visible in the DesktopDatePicker's input field even when the field is unfocused; consistent with the behaviour of the DateField (https://mui.com/x/react-date-pickers/date-field/).

### Context 🔦

Trying to use the component in a production setting. This behaviour results in an unintuitive and jarring user experience.
I haven't been able to find a workaround yet, so any advice on what I might be able to do in the meantime is much appreciated.

### Your environment 🌎

Issue can be reproduced in the examples in the MUI docs: https://mui.com/x/react-date-pickers/date-picker/
Using latest version of Chrome at time of posting (Version 111.0.5563.65).

### Order ID or Support key 💳 (optional)

_No response_

### PR DESCRIPTION
**Pull request by flaviendelangle**
Fixes #8296 
Fixes a bug caused by #8287 (not released yet) and #8082

If you clean a section in the field, it now cleans the entire field when used inside a picker ?

Why ? Because the picker replaces `Invalid Date` by `null` in its internal state, to when you clean a section, the field fires `onChange` with the invalid date (#8082), the picker replaces it with `null` and the field updates the sections based on this new value (the fix from #8287 uncovered this broken behavior).

____

The views don't like invalid dates.
That's something we should probably improve at some point.
For now, I just allows to store `Invalid Date` in the picker state and instead replace it with `null` before passing it to the views.
That way the field does receive the `Invalid Date` it asked for.
I also changed the behavior of `valueManager.areValuesEqual` so that two invalid dates are considered equal (otherwise we have an infinite loop).

### CODE REVIEW

File: packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts [hunk 1]
```diff
@@ -417,8 +412,6 @@ export const usePickerValue = <
     setDate({
       value: valueManager.emptyValue,
       action: 'acceptAndClose',
-      // force `onChange` in cases like input (value) === `Invalid date`
-      forceOnChangeCall: !valueManager.areValuesEqual(utils, value as any, valueManager.emptyValue),
```
Comment (LukasTy) [inline packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts:421]: 🙌 The new solution for these cases seems way nicer! 👍 
Can we now ditch this prop/flag from the `setDate`? 🤔

**Review by LukasTy — APPROVED** [review]
LGTM! Nice and clean work. 👍

**Comment by flaviendelangle** [comment]
> P.S. Should we update the label to regression instead of a bug? thinking

I used `bug` because there is no regression in the published packages.
Would you rather always use `regression` even when the regression is only on the `master` branch ?

File: packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts (hunk 1 above)
Comment (flaviendelangle) [reply packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts:421]: That was a nice side effect :ok_hand: 
I'll clean it :+1:

**Review by alexfauquette — APPROVED** [review]

**Comment by LukasTy** [comment]
> I used `bug` because there is no regression in the published packages.

Isn't #8296 the regression that is caused by this same behavior? 🤔

**Comment by flaviendelangle** [comment]
:exploding_head: it solves it indeed
It's not the fix I had in mind, but if it works... :laughing: 

By the way, I think this expression should not have `!state.value`:

```tsx
  const shouldShowPlaceholder =
    !inputHasFocus &&
    (!state.value || valueManager.areValuesEqual(utils, state.value, valueManager.emptyValue));
```

**Pull request merged**